File: output/java/Easy/13-add-binary.java
/**https://leetcode.com/problems/add-binary/ */
//Given two binary strings a and b, return their sum as a binary string.


class AddBinary {
    public String addBinary(String a, String b) {
        int carry = 0;
        int i = a.length() - 1;
        int j = b.length() - 1;
        StringBuilder sb = new StringBuilder();
        while(i >= 0 || j >= 0){
            int sum = carry;
            if(i >= 0){
                sum += a.charAt(i) - '0';
                i--;
            }
            if(j >= 0){
                sum += b.charAt(j) - '0';
                j--;
            }
            sb.append(sum % 2);
            carry = sum / 2;
        }
        if(carry == 1){
            sb.append(1);
        }
        return sb.reverse().toString();
        
    }
}

    

File: output/py/Easy/13-add-binary.py
#https://leetcode.com/problems/add-binary/
#Given two binary strings a and b, return their sum as a binary string.


class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        res=""
        carry=0
        i=len(a)-1
        j=len(b)-1
        while i>=0 or j>=0:
            if i>=0 and j>=0:
                res=str((int(a[i])+int(b[j])+carry)%2)+res
                carry=(int(a[i])+int(b[j])+carry)/2
            elif i>=0:
                res=str((int(a[i])+carry)%2)+res
                carry=(int(a[i])+carry)/2
            else:
                res=str((int(b[j])+carry)%2)+res
                carry=(int(b[j])+carry)/2
            i=i-1
            j=j-1
        if carry==1:
            res="1"+res
        return res

    